### PR TITLE
Refactor how we build up the list of forms page

### PR DESCRIPTION
**Pull request by aliuk2012**
### What problem does this pull request solve?

Before trying to display "created by" column, it felt like a good opportunity to refactor our existing implementation for a number of reasons.

- The index view is doing an awful lot and has a lot of logic it in it (e.g looping through each form and deciding which draft/live url to use and which status to show for a form. This makes it extremely difficult to test even in a view spec with all the various states that the view could be in.
- The current implementation would mean that the view might have to query each form "created by" user as its iterating on it  which would be slow

Trello card: https://trello.com/c/ePWiTjCg/1260-show-which-user-made-each-form

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: spec/service/form_list_service_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        expect(service.data[:head].first).to eq I18n.t("home.form_name_heading")
+      end
+
+      it "contains 'Status' column heading and is numeric " do
+        expect(service.data[:head].last).to eq text: I18n.t("home.form_status_heading"), numeric: true
+      end
+    end
+
+    describe "rows" do
+      it "has a row for each form passed to the service" do
```
Comment (danworth) [inline spec/service/form_list_service_spec.rb:48]: Just wondering if there's any merit in a test case for when there are no forms present? Is that covered in a request spec or view spec somewhere, wondering how `govuk_table` handles if `data.rows` is empty, just thinking about edge-cases etc.

**Review by danworth — APPROVED** [review]
This looks like a nice refactor 🏅 , the service is simpler to test and the view is much simplified. I've run it locally and in addition to CI passing this changes look good.

File: app/service/form_list_service.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# require "helpers/govuk_rails_compatible_link_helper"
+class FormListService
+  include Rails.application.routes.url_helpers
+  include ActionView::Helpers::UrlHelper
+  include GovukRailsCompatibleLinkHelper
+
+  attr_accessor :forms, :current_user, :search_form
+
+  class << self
+    def call(**args)
```
[the remark this answers is not in the document]
Comment (aliuk2012) [reply app/service/form_list_service.rb:10]: Its not really a Rails idiom but more of a design pattern called Factory Method pattern. There are other ways to write this sort of thing

```
class Hello
  def self.call(**args)
    new(**args)
  end
end
```
The benefit of using `class << self` is that you dont have to add `self` prefix to each method  inside that call. 
In terms of services it feels easier to `call` a service than to create `new`  services which is why you mostly see this pattern being used for our services but not forms_objects as an example.
